src/aiodynamo/client.py

```diff
@@ -220,45 +234,64 @@ def table(self, name: TableName) -> Table:
     ):
-        lsis: List[LocalSecondaryIndex] = lsis or []
-        gsis: List[GlobalSecondaryIndex] = gsis or []
-        stream = stream or StreamSpecification()
-        attributes = {}
-        attributes.update(keys.to_attributes())
-        for index in chain(lsis, gsis):
-            attributes.update(index.schema.to_attributes())
+        attributes = keys.to_attributes()
+        if lsis is not None:
+            for index in lsis:
+                attributes.update(index.schema.to_attributes())
+        if gsis is not None:
+            for index in gsis:
+                attributes.update(index.schema.to_attributes())
         attribute_definitions = [
             {"AttributeName": key, "AttributeType": value}
             for key, value in attributes.items()
         ]
-        key_schema = keys.encode()
-        local_secondary_indexes = [index.encode() for index in lsis]
-        global_secondary_indexes = [index.encode() for index in gsis]
-        provisioned_throughput = throughput.encode()
-        stream_specification = stream.encode()
-        await self.core.create_table(
-            **clean(
-                AttributeDefinitions=attribute_definitions,
-                TableName=name,
-                KeySchema=key_schema,
-                LocalSecondaryIndexes=local_secondary_indexes,
-                GlobalSecondaryIndexes=global_secondary_indexes,
-                ProvisionedThroughput=provisioned_throughput,
-                StreamSpecification=stream_specification,
-            )
-        )
-        return None
-
-    async def enable_time_to_live(self, table: TableName, attribute: str):
-        await self.core.update_time_to_live(
-            TableName=table,
-            TimeToLiveSpecification={"AttributeName": attribute, "Enabled": True},
+        payload = {
+            "AttributeDefinitions": attribute_definitions,
+            "TableName": name,
+            "KeySchema": keys.encode(),
+            "ProvisionedThroughput": throughput.encode(),
+        }
+        if lsis:
+            payload["LocalSecondaryIndexes"] = [index.encode() for index in lsis]
+        if gsis:
+            payload["GlobalSecondaryIndexes"] = [index.encode() for index in gsis]
+        if stream:
+            payload["StreamSpecification"] = stream.encode()
+
+        await self.send_request(action="CreateTable", payload=payload)
+        if wait_for_active:
+            if not isinstance(wait_for_active, WaitConfig):
+                wait_for_active = WaitConfig.default()
+            attempts = 0
+            while attempts < wait_for_active.max_attempts:
+                try:
+                    description = await self.describe_table(name)
+                    if description.status == TableStatus.active:
+                        return
+                except TableNotFound:
+                    pass
+                attempts += 1
+                await asyncio.sleep(wait_for_active.retry_delay)
```

Comment:
I wonder if an async retry iterator (helper) could be implemented in the `WaitConfig`...
I mean if it can be implemented in a way that's not too ugly... 🤔

Reply:
I tried a generic `wait` implementation, but the exception handling etc got very ugly very quick, so I resorted to this solution. I'm not happy with it but can't come up with a sane API.